nav2_dwb_controller/dwb_core/src/publisher.cpp

```diff
@@ -214,7 +214,9 @@ DWBPublisher::publishTrajectories(const dwb_msgs::msg::LocalPlanEvaluation & res
     }
     ma.markers.push_back(m);
   }
-  marker_pub_->publish(ma);
+  if (node_->count_subscribers(marker_pub_->get_topic_name()) > 0) {
```

Comment:
There's a bunch of "work" happening above this, can you move the subscriber count to the start of the method and just skip everything if no one is listening?